Source/Core/Core/IOS/ES/Formats.cpp

```diff
@@ -596,20 +597,20 @@ std::vector<std::array<u8, 20>> SharedContentMap::GetHashes() const
 
 std::string SharedContentMap::AddSharedContent(const std::array<u8, 20>& sha1)
 {
-  auto filename = GetFilenameFromSHA1(sha1);
-  if (filename)
-    return *filename;
+  if (auto filename = GetFilenameFromSHA1(sha1))
+    return *std::move(filename);
 
-  const std::string id = fmt::format("{:08x}", m_last_id);
-  Entry entry;
-  std::copy(id.cbegin(), id.cend(), entry.id.begin());
+  Entry& entry = m_entries.emplace_back();
+  // `entry.id` is a std::array capable of storing every nibble of `m_last_id`.
+  static_assert(std::tuple_size_v<decltype(entry.id)> == sizeof(m_last_id) * 2);
```

Comment:
Maybe also add an assert that `sizeof(m_last_id) == 4` (or equivalent)? Padding to 8 digits only makes sense if this is the case.